src/scripts/core/models/grid.ts

```diff
@@ -6,24 +6,16 @@ import { Token } from "./token";
 export class Grid extends Container {
 
     private gridSize: number;
-    private columns: Column[];
-    private activeToken: number;
-
+    private columns: Column[] = [];
+    private tokensSelected: number = 0;
+    private selectedTokens: [Token | undefined, Token | undefined] = [undefined, undefined];
```

Comment:
could this not just be a Token[] which you change the length of?
this would mean not having to have undefined as part of the typing

to get tokensSelected you could instead call selectedTokens.length